### ISSUE TITLE
6.1.1 build fails because of several unused functions (and a few more errors)

### ISSUE THREAD

**Issue by dellelce** [varnishcache/varnish-cache#2808]
## Expected Behavior

build succeeds

## Current Behavior

build fails (was successful until 6.1.0)

## Possible Solution

 1. remove -Werror=unused-function
 2. or: remove unused functions
 3. or: use them someway :-)

## Steps to Reproduce (for bugs)

1. "normal" build options, see log: https://travis-ci.org/dellelce/mkit/jobs/446700513
   original script:
     https://github.com/dellelce/mkit/blob/master/mkit.components.sh#L692
    https://github.com/dellelce/mkit/blob/master/mkit.lib.sh#L360

## Your Environment

* Version used: 6.1.1
* Operating System and version: alpine inside docker 
* Source of binary packages used (if any)

**Comment by dellelce** [comment]
never look at a log file from the end.

**Comment by hermunn** [comment]
Starting on line 1037 in the log, I see the following

```
make[4]: Entering directory '/app-build/mkit_workdir/build_1352_261018/varnish/bin/varnishtest'
awk -f /app-build/mkit_workdir/src_1352_261018/varnish/varnish-6.1.1/bin/varnishtest/gensequences /app-build/mkit_workdir/src_1352_261018/varnish/varnish-6.1.1/bin/varnishtest/sequences \
    > ./teken_state.h
  CC       varnishtest-teken.o
In file included from /app-build/mkit_workdir/src_1352_261018/varnish/varnish-6.1.1/bin/varnishtest/teken.c:736:0:
./teken_state.h:3:22: error: expected identifier or '(' before numeric constant
 static teken_state_t 8;
```
The rest of the errors should probably be ignored until this one is fixed.

The source build fine on our supported platforms, but we would love to have it work more. Can you check if this is a problem unique to 6.1.1, or if it also in *master*?

**Comment by gquintard** [comment]
can you share the teken.h file please?

**Comment by dellelce** [comment]
@hermunn, I just tried  from master (ee33662a162cbc4fcc7fb8a93d143f85b7786eae) and it builds fine.

@gquintard : I am using the `teken.h` that comes with 6.1.1 not a custom one.

**Comment by gquintard** [comment]
sorry, meant `teken_state.h`, but from what I'm seeing, it's the same as the one on other platforms

**Comment by gquintard** [comment]
0723fa9f5d7e8856b05031a33040ef71e95dd101 broke it, haven't dug more yet

**Comment by gquintard** [comment]
@hermunn, looks like the latest update to `gensequence` undid 9f50c4bd12303674890ca742ce6b9592a29e3ba9

I leave it to you to cherry-pick if you feel that's needed

**Comment by hermunn** [comment]
@gquintard My bad, sorry.

@dellelce Can you test https://github.com/hermunn/varnish-cache/tree/fix-2808 just to be sure? If it this is enough to fix it, then I will push it to 6.1.

**Comment by dellelce** [comment]
@hermunn: It builds fine:

```bash
bash-4.4# cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.8.1
PRETTY_NAME="Alpine Linux v3.8"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"
bash-4.4# ldd varnishd
        /lib/ld-musl-x86_64.so.1 (0x7f9761bc7000)
        libpcre.so.1 => /tmp/varnish/lib/libpcre.so.1 (0x7f9761634000)
        libexecinfo.so.1 => /usr/lib/libexecinfo.so.1 (0x7f9761424000)
        libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f9761bc7000)
bash-4.4# ./varnishd -V
varnishd (varnish-6.1.1 revision NOGIT)
Copyright (c) 2006 Verdens Gang AS
Copyright (c) 2006-2015 Varnish Software AS
```
